### PR TITLE
Create Rule “software-engineers-should-only-roll-forward/rule”

### PR DESCRIPTION
**Pull request by ozairashfaqueSSW**
Automatically generated by Decap CMS

### CODE REVIEW

**Review by tiagov8 — CHANGES_REQUESTED** [review]
Hey @ozairashfaqueSSW 

- [x] 1. Please add new rule to a category

**Comment by ozairashfaqueSSW** [comment]
> Hey @ozairashfaqueSSW
> 
> * [x]  1. Please add new rule to a category



I have updated the category `Rules to Better DevOps` with the new rule `Do you know when to Roll Forward and when to Roll Backward?`

File: rules/do-you-know-roll-forward-or-roll-backward/rule.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+type: rule
+tips: ""
+title: Do you know when to Roll Forward and when to Roll Backward?
+seoDescription: Learn when to roll forward and when to roll backward in software deployment.
+uri: do-you-know-roll-forward-or-roll-backward
```
Comment (wicksipedia) [inline rules/do-you-know-roll-forward-or-roll-backward/rule.md:6]:
```suggestion
uri: when-to-roll-forward-or-roll-backward
```

**Comment by wicksipedia** [comment]
real developers roll forward

ideally your delivery pipeline should catch catastrophic releases early on so that you don't need to roll back - especially with database changes (and potential data-loss), you need to take more things into consideration in order to support rolling back

fyi the rule feels very AI generated

**Review by matt-goldman-ssw — CHANGES_REQUESTED** [review]
I think this rule needs to be redrafted. We can probably dispense with the whole "when to roll forward" section and instead position the rule as saying you should always roll forward unless specifically unable to do so. And here are some examples of why you may not be able to continue forward.

**Comment by JackDevAU** [comment]
**Don't forget to also fix the `uri` to match the folder name!** 

Make sure the GItHub Action: `Check-Folder-Name / rename-folder ` is  passing ✅

`rules/do-you-know-only-roll-forward`
!=
`uri: software-engineers-should-only-roll-forward`

**Review by matt-goldman-ssw — COMMENTED** [review]
LGTM

File: rules/do-you-know-only-roll-forward/rule.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+type: rule
+tips: ""
```
[the remark this answers is not in the document]
Comment (tino-liu) [reply rules/do-you-know-only-roll-forward/rule.md]: Resolved

**Review by matt-goldman-ssw — APPROVED** [review]

**Review by wicksipedia — APPROVED** [review]

**Pull request merged**